[WTF] Make Assertion inlined

Constellation opened the pull request:
#### c5e2f6fceb7fa508c026286a412a8b3b01c7f0c0
<pre>
[WTF] Make Assertion inlined
<a href="https://bugs.webkit.org/show_bug.cgi?id=270952">https://bugs.webkit.org/show_bug.cgi?id=270952</a>
<a href="https://rdar.apple.com/124569636">rdar://124569636</a>

Reviewed by Keith Miller.

This inlined function with optnone is originally introduced in 203263@main to fix internal build failures.
But these affected code no longer exists. This patch makes WTFCrashWithInfo completely inlined function with
appropriate inline asm so that we can make caller of them optimized.

* Source/WTF/wtf/Assertions.h:

Canonical link: <a href="https://commits.webkit.org/276095@main">https://commits.webkit.org/276095@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5f5faac4781c47fe1d333ae02672a6d533ddd79b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43706 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22754 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46145 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46349 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39814 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26591 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20162 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36097 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44280 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19768 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37643 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17079 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17307 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38711 "Found 1 new test failure: imported/w3c/web-platform-tests/xhr/setrequestheader-case-insensitive.htm (failure)") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1763 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/37149 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39859 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39015 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47905 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43328 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18739 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15320 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42899 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20148 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41581 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20339 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50338 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5975 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19774 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10148 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/WTF/wtf/Assertions.cpp:
```diff
@@ -750,6 +750,12 @@ void WTFCrashWithInfoImpl(int, const char*, const char*, int, uint64_t reason)
+void WTFCrashWithInfo(int, const char*, const char*, int)
+{
+    __asm__ volatile (CRASH_INST : : );
+    __builtin_unreachable();
```
From talking to clang folks apparently we should use `__builtin_trap()` because `__builtin_unreachable()` is technically a signal to the compiler that this basic block can be removed.

Constellation replied:
Sounds good. Changed.

kmiller68 reviewed (approved):
r=me with request

kmiller68 commented on Source/WTF/wtf/Assertions.h:
```diff
@@ -248,16 +248,58 @@ WTF_EXPORT_PRIVATE void WTFReleaseLogStackTrace(WTFLogChannel*);
 
 WTF_EXPORT_PRIVATE bool WTFIsDebuggerAttached(void);
 
+#if (OS(DARWIN) || PLATFORM(PLAYSTATION)) && (CPU(X86_64) || CPU(ARM64))
```
I think this `#if` is messing up the wpe build. I don't think we need it anyway.

kmiller68 reviewed (approved):
r=me with build fix.

Constellation committed:
[WTF] Make Assertion inlined
https://bugs.webkit.org/show_bug.cgi?id=270952
rdar://124569636

Reviewed by Keith Miller.

This inlined function with optnone is originally introduced in 203263@main to fix internal build failures.
But these affected code no longer exists. This patch makes WTFCrashWithInfo completely inlined function with
appropriate inline asm so that we can make caller of them optimized.

* Source/WTF/wtf/Assertions.h:

Canonical link: https://commits.webkit.org/276095@main
